examples_extra/ASDKgram-Swift/ASDKgram-Swift/PhotoFeedModel.swift

```diff
@@ -67,7 +67,9 @@ final class PhotoFeedModel {
 	private func fetchNextPageOfPopularPhotos(replaceData: Bool, numberOfAdditionsCompletion: @escaping (Int, NetworkingErrors?) -> ()) {
 
 		if currentPage == totalPages, currentPage != 0 {
-			return numberOfAdditionsCompletion(0, .customError("No pages left to parse"))
+			DispatchQueue.main.async {
+				return numberOfAdditionsCompletion(0, .customError("No pages left to parse"))
+			}
```

Comment:
You need to add a "return" here otherwise it would execute this path although it should return like it did before.